pkg/storage/storage.go

```diff
@@ -483,39 +615,17 @@ func (s *storageModule) createSubvol(size uint64, name string, poolType pkg.Devi
 			continue
 		}
 
-		candidates = append(candidates, Candidate{
+		candidates = append(candidates, candidate{
```

Comment:
This doesn't sound right. You should only add the disk to the candidates list ONLY if the disk has enough space for the requested size. So if the disk doesn't have enough space for that subvolume, it shouldn't be considered a candidate.

Also in the older code, the candidates where sorted so the one with the most disk space available is at the head of the candidate list so it gets to be tried first.

Reply:
This is already implemented? https://github.com/threefoldtech/zos/blob/93ff52177cd6cfc3292cef81e181cf9e3ce088d7/pkg/storage/storage.go#L603-L611